src/common/grpc-expr/src/alter.rs

```diff
@@ -124,11 +130,17 @@ pub fn create_table_schema(expr: &CreateTableExpr) -> Result<RawSchema> {
     Ok(RawSchema::new(column_schemas))
 }
 
+pub fn create_table_schema(expr: &CreateTableExpr) -> Result<RawSchema> {
+    create_table_schema_options(expr, true)
+}
+
 pub fn create_expr_to_request(
     table_id: TableId,
     expr: CreateTableExpr,
 ) -> Result<CreateTableRequest> {
-    let schema = create_table_schema(&expr)?;
+    let require_time_index = expr.engine != common_catalog::consts::IMMUTABLE_FILE_ENGINE;
```

Comment:
suggestion: Make require_time_index an argument of this function, otherwise there is no difference from checking the engine in `create_table_schema()`. 

The `common-grpc-expr` should not be aware of whether an engine requires a time index. Let's move this check to places invoking create_expr_to_request().